tests/diagnostics/uninit-var.slang

```diff
@@ -0,0 +1,35 @@
+// uninit-var.slang
+//TEST:SIMPLE(filecheck=CHECK): -target hlsl -entry computeMain -profile cs_6_0
```

Comment:
I think there should also be a test for the opposite scenario: we do not throw a warning/error if the same scenario has an initialized variable (easier to debug and catch if something breaks)

Todo this, add something like the following:
```
//TEST:SIMPLE(filecheck=CHECK_INITIALIZED): -target hlsl -entry computeMain -profile cs_6_0 -DINITIALIZED
// ...
//CHECK_INITIALIZED-NOT: error 41015
#ifdef INITIALIZED
    float uninitializedVariable = 0;
#else 
    float uninitializedVariable;
#endif
```

Reply:
noted. i added this case to the test.